### PR TITLE
New Record: Track_3_optimization: Add Contra-Muon 3225 steps

### PR DESCRIPTION
**Pull request by nilin**
# Track 3: Add Contra-Muon to previous leader (built on PR 274)

Builds on PR 274 and adds Contra-Muon (https://github.com/nilin/contra-muon), reaching <3.28 at 3225 steps with p<<1e-3. Mean val at 3200 steps is `3.2793`, but we don't have the statistical power for the stronger claim of 3200 steps.

## Contra-Muon
Contra-Muon modifies the Muon Newton-Schulz momentum update by subtracting `CONTRA_MUON / 
2` times the operator-normalized momentum gradient. See https://github.com/nilin/contra-muon for an explanation of the reasoning behind this definition. In these runs, `CONTRA_MUON=0.4`, so the subtracted component is `0.2 * normalized_grad`. That is:
```
update = muon_update - contra_muon / 2 * operator_normalized_momentum_gradient
```
Further postprocessing is retained from PR 274, see below.

## Ancestry
The rest of the optimizer follows the PR 274 NorMuon-lite setup (https://github.com/KellerJordan/modded-nanogpt/pull/274). None of the existing hyperparameters were changed.
- per-row/column variance normalization on the Newton-Schulz update
- u/w-floor postprocessing with target `0.35`
- Muon lr `0.0375`

## Note on stale parameter in logs
**The archived scripts/logs print `muon_weight_decay=0.025`, but `Muon.step` does not use the stored `weight_decay` param-group value. The effective Muon weight decay for these runs is therefore zero. The script is left unchanged to stay consistent with the submitted logs. The inactive logged parameter is documented in the submission directory's README.md**

## Results

- mean val loss at 3200 steps: `3.2793`
- mean val loss at 3225 steps: `3.2785`
We ran with `n=16` non-cherry-picked seeds 1-16. The track 3 statistic for 3225 steps is: `(3.28 - mean) * sqrt(n) = 0.00583752` which satisfies the requirement. One-sided z-test with `sigma=0.0013`: `z=4.4904`, `p=3.55e-6`.

| Seed | 3200 val | 3225 val | 
| -: | -: | -: | 
| Mean | 3.27929188 | 3.27854062 | 
| 1 | 3.27699 | 3.27626 | 
| 2 | 3.27827 | 3.27752 | 
| 3 | 3.28074 | 3.28000 | 
| 4 | 3.27918 | 3.27846 | 
| 5 | 3.28046 | 3.27967 | 
| 6 | 3.28063 | 3.27989 | 
| 7 | 3.27907 | 3.27832 | 
| 8 | 3.27947 | 3.27867 | 
| 9 | 3.27880 | 3.27807 | 
| 10 | 3.27981 | 3.27906 | 
| 11 | 3.27989 | 3.27912 | 
| 12 | 3.27808 | 3.27733 | 
| 13 | 3.27926 | 3.27851 | 
| 14 | 3.27951 | 3.27880 | 
| 15 | 3.27869 | 3.27791 | 
| 16 | 3.27982 | 3.27906 | 

The logs include the full self-embedded training script source, satisfying the
reproducibility requirement.

## Ablation

To verify that Contra-Muon yields improvement over the ancestor script, apply the Welch t-test at step 3125, the last validation step that the two entries have in common. We observe improvement with significance `p<0.01`:

```
PR274 (ancestor), n=8:
at step 3125
mean = 3.28664
sample sd = 0.00095
standard error = 0.00034

PR275 (this), n=16:
at step 3125
mean = 3.28552
sample sd = 0.00105
standard error = 0.00026

difference in means = 0.00111
Welch SE of difference = 0.00043
t = 2.6135
p < 0.01
```

Welch's t-test confirms that the modification yields improvement with `p<0.01`. This test is not a requirement but is included to confirm that this is not validation hacking. Note that none of the existing hyperparameters were tuned.

### CODE REVIEW

**Comment by alint77** [comment]
I tried running the NS at fp32, and also replacing NS5 with PolarExpress from the paper, both resulted in worse loss surprisingly

**Comment by KellerJordan** [comment]
LGTM!
